src/app/InteractionModelEngine.cpp

```diff
@@ -305,6 +308,58 @@ Status InteractionModelEngine::OnInvokeCommandRequest(Messaging::ExchangeContext
     return Status::Success;
 }
 
+CHIP_ERROR InteractionModelEngine::ParseAttributePaths(Access::SubjectDescriptor & aSubjectDescriptor,
+                                                       AttributePathIBs::Parser & aAttributePathListParser,
+                                                       bool & aHasValidAttributePath, size_t & aRequestedAttributePathCount)
+{
+    TLV::TLVReader pathReader;
+    aAttributePathListParser.GetReader(&pathReader);
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    aHasValidAttributePath       = false;
+    aRequestedAttributePathCount = 0;
+
+    while (CHIP_NO_ERROR == (err = pathReader.Next()))
+    {
+        VerifyOrReturnError(TLV::AnonymousTag() == pathReader.GetTag(), CHIP_ERROR_INVALID_TLV_TAG);
+
+        AttributePathIB::Parser path;
+        ReturnErrorOnFailure(path.Init(pathReader));
+
+        AttributePathParams params;
+        ReturnErrorOnFailure(path.ParsePath(params));
+
+        //
+        // We create an iterator to point to a single item object list that tracks the path we just parsed.
+        // This avoids the 'parse all paths' approach that is employed in ReadHandler since we want to
+        // avoid allocating out of the path store during this minimal initial processing stage.
+        //
+        ObjectList<AttributePathParams> paramsList;
+        paramsList.mValue = params;
```

Comment:
Can we not parse the path directly into paramsList.mValue so we use up less stack space and don't have to copy the path?

Reply:
yes, updated, thanks